### PR TITLE
Update `Tabs` usage guidelines

### PR DESCRIPTION
**Pull request by cveigt**
### :pushpin: Summary

This branch adds usage examples and behavior guidelines for the `Tabs` component.

Preview 👉 https://hds-website-git-cv-update-tabs-docs-hashicorp.vercel.app/components/tabs

### :link: External links

Jira ticket: [HDS-737](https://hashicorp.atlassian.net/browse/HDS-737)

***
:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

[HDS-737]: https://hashicorp.atlassian.net/browse/HDS-737?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: website/docs/components/tabs/partials/guidelines/guidelines.md [hunk 1]
```diff
@@ -8,6 +8,46 @@
...
+!!! Do
+
+Use Tabs to break up content that is related to each other and is on the same hierarchical level.
+
+![Tab usage example](/assets/components/tabs/tab-usage-example-01.png)
+!!!
+
+!!! Dont
+
+Don't use Tabs to take the user to a new URL or as a primary or secondary navigation.
```
Comment (MelSumner) [inline website/docs/components/tabs/partials/guidelines/guidelines.md:22]: FWIW, a query param is technically a new URL, and that's how we use our tabs component in our own docs (i.e., https://helios.hashicorp.design/components/modal?tab=code) 

I wonder if there's a different way to word this?

Comment (heatherlarsen) [reply website/docs/components/tabs/partials/guidelines/guidelines.md]: While the Helios website tabs and the HDS tabs differ, I addressed this in another round of iterations.

**Review by jorytindall — APPROVED** [review]
:shipit:

File: website/docs/components/tabs/partials/guidelines/guidelines.md [hunk 2]
```diff
@@ -2,11 +2,50 @@
...
 ### When not to use
 
-- For navigation, consider navigation, [Standalone Links](/components/link/standalone), or [Breadcrumb](/components/breadcrumb).
+- For navigation, consider [SideNav](/components/side-nav), [Standalone Links](/components/link/standalone), or [Breadcrumb](/components/breadcrumb).
+- As a stepper or progress bar for multi-step workflows, consider a stepper, using the [Stepper Indicator](/components/stepper).
+- To see multiple content areas at the same time or to fully collapse the content area, use [Accordion](/components/accordion).
+
+!!! Dont
+
+Don't use Tabs for primary or secondary navigation.
```
Comment (KristinLBradley) [inline website/docs/components/tabs/partials/guidelines/guidelines.md:22]:
```suggestion
Don’t use Tabs for primary or secondary navigation.
```

File: website/docs/components/tabs/partials/guidelines/guidelines.md [hunk 3]
```diff
@@ -2,11 +2,50 @@
...
+
+![Tab usage example](/assets/components/tabs/tab-usage-example-02.png)
+!!!
+
+!!! Dont
+
+Don't use Tabs for sequential content.
```
Comment (KristinLBradley) [inline website/docs/components/tabs/partials/guidelines/guidelines.md:29]:
```suggestion
Don’t use Tabs for sequential content.
```

File: website/docs/components/tabs/partials/guidelines/guidelines.md [hunk 4]
```diff
@@ -2,11 +2,50 @@
...
+
+![Tab usage example](/assets/components/tabs/tab-usage-example-03.png)
+!!!
+
+## Tab best practices
+
+!!! Dont
+
+We don't recommend using a singular Tab; instead consider using a heading.
```
Comment (KristinLBradley) [inline website/docs/components/tabs/partials/guidelines/guidelines.md:38]:
```suggestion
We don’t recommend using a singular Tab; instead consider using a heading. 
```

File: website/docs/components/tabs/partials/guidelines/guidelines.md [hunk 5]
```diff
@@ -2,11 +2,50 @@
...
+!!! Dont
+
+We don't recommend using a singular Tab; instead consider using a heading. 
+
+![Tabs with only one tab](/assets/components/tabs/tab-behavior-single-button.png)
+!!!
+
+!!! Dont
+
+Don't trigger a Tab change via an external action, such as a submit or next button. Tab changes should only be triggered by the Tab itself as that is the expected user interaction.
```
Comment (KristinLBradley) [inline website/docs/components/tabs/partials/guidelines/guidelines.md:45]:
```suggestion
Don’t trigger a Tab change via an external action, such as a submit or next button. Tab changes should only be triggered by the Tab itself as that is the expected user interaction. 
```

**Review by KristinLBradley — APPROVED** [review]
Just added a few minor suggested punctuation changes.

**Pull request merged**